R/pkg/R/DataFrame.R

```diff
@@ -1138,6 +1138,9 @@ setMethod("collect",
                   if (!is.null(PRIMITIVE_TYPES[[colType]]) && colType != "binary") {
                     vec <- do.call(c, col)
                     stopifnot(class(vec) != "list")
+                    # If vec is an vector with only NAs, the type is logical
```

Comment:
Yes. My first commit was trying to cast the column to its corresponding R data type explicitly, even if it is an vector with all NAs. However some existed tests were failed and expecting to get logical NA. For example
```
3. Failure: column functions (@test_sparkSQL.R#1280) ---------------------------
collect(select(df, first(df$age)))[[1]] not equal to NA.
Types not compatible: double vs logical
4. Failure: column functions (@test_sparkSQL.R#1282) ---------------------------
collect(select(df, first("age")))[[1]] not equal to NA.
Types not compatible: double vs logical
```